cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -218,7 +217,7 @@ public void create(HttpRequest request, HttpResponder responder,
     StreamId streamId;
     try {
       // Verify stream name
-      streamId = Id.Stream.from(namespaceId, stream).toEntityId();
+      streamId = new StreamId(namespaceId, stream);
```

Comment:
This seems inconsistent than other handlers (wrapping in try-catch and rethrowing as BadRequestException).
Should be made consistent, or at least make sure the response code from the other handlers is appropriate, when a malformed name is provided.